src/main/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClient.java

```diff
@@ -617,16 +617,22 @@ public SendMessageBatchResponse sendMessageBatch(SendMessageBatchRequest sendMes
 
         List<SendMessageBatchRequestEntry> batchEntries = new ArrayList<>(sendMessageBatchRequest.entries().size());
 
+        boolean hasS3Entries = false;
         for (SendMessageBatchRequestEntry entry : sendMessageBatchRequest.entries()) {
             //Check message attributes for ExtendedClient related constraints
             checkMessageAttributes(entry.messageAttributes());
 
             if (clientConfiguration.isAlwaysThroughS3() || isLarge(entry)) {
                 entry = storeMessageInS3(entry);
+                hasS3Entries = true;
             }
             batchEntries.add(entry);
         }
 
+        if (hasS3Entries) {
```

Comment:
Good catch, do we actually need this `hasS3Entries `? Since `batchEntries` will contain the actual entries we want to send (some might be s3 pointers and some might be the actual message) we can update the entries for the `sendMessageBatchRequest` every-time.

Reply:
I guess just to avoid having to copy the new entries.

Reply:
thanks for quick response! and for merging the PR :D